connectors/sources/tests/test_github.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the Github source class methods"""
+from http import HTTPStatus
+from unittest import mock
+
+import pytest
+from gidgethub import BadRequest
+
+from connectors.source import ConfigurableFieldValueError, DataSourceConfiguration
+from connectors.sources import github
+from connectors.sources.github import GitHubDataSource
+from connectors.sources.tests.support import create_source
+from connectors.tests.commons import AsyncIterator
+
+
+def test_get_configuration():
+    config = DataSourceConfiguration(GitHubDataSource.get_default_configuration())
+
+    assert config["repositories"] == ["*"]
+    assert config["github_token"] == "changeme"
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("field", ["repositories", "github_token"])
+async def test_validate_config_missing_fields_then_raise(field):
+    source = create_source(GitHubDataSource)
+    source.configuration.set_field(name=field, value="")
+
+    with pytest.raises(ConfigurableFieldValueError):
+        await source.validate_config()
+
+
+@pytest.mark.asyncio
+async def test_ping_with_successful_connection():
+    source = create_source(GitHubDataSource)
+    source.github_client._get_session.getitem = mock.AsyncMock(
+        return_value={"user": "username"}
+    )
+    await source.ping()
+
+
+@pytest.mark.asyncio
+async def test_ping_with_unsuccessful_connection():
+    source = create_source(GitHubDataSource)
+
+    with mock.patch.object(
+        source.github_client,
+        "ping",
+        side_effect=Exception("Something went wrong"),
+    ):
+        with pytest.raises(Exception):
+            await source.ping()
+
+
+@pytest.mark.asyncio
+async def test_close_without_session():
+    source = create_source(GitHubDataSource)
+    await source.close()
+    source.github_client._get_session
```

Comment:
What's the purpose of this line? Do you mean `source.github_client._get_session()`. We should also assert something here, i.e. that the correct closing methods were called.

Reply:
Yes to get the client we called `_get_session` and as it is `cached_property` and returns github client we did not call like this `source.github_client._get_session()`